### ISSUE TITLE
R2_Score() is inconsistent with other methods

### ISSUE THREAD

**Issue by MaurizioSicorello** [yanyachen/MLmetrics#13]
When trying to compute R-Squared from predicted and actual values, the results from R2_Score() did not match other methods. (they were very large and negative).

Here is some R-Code to reproduce this issue: 

`x = rnorm(20)
y = rnorm(20)

test = lm(y~x)
summary(test)
y_pred = predict(test)

# MLmetrics
R2_Score(y, y_pred)

# squared correlation
cor(y_pred, y)^2

# sums of squares formula
PRESS = sum((y - y_pred)^2)
SS = sum((y - mean(y))^2)
1 - (PRESS/SS)`

**Comment by hadjipantelis** [comment]
The example presented uses the wrong order of input arguments.

**Comment by fionag11** [comment]
It's still wrong. I have also noticed this error.  Nonsensical negative values alerted me something was wrong. In fact, sometimes positive values are correct, and sometimes they are way off as well.

**Comment by jmcastagnetto** [comment]
I just checked the R2_Score function definition in https://github.com/yanyachen/MLmetrics/blob/91eb6704272431d26db42468b5b6eed658f4d903/R/Regression.R#L217 and the example proposed in this issue, but I cannot see the errors mentioned. 

I used the code below:

```r
R2_Score <- function(y_pred, y_true) {
  R2_Score <- 1 - sum((y_true - y_pred)^2) / sum((y_true - mean(y_true))^2)
  return(R2_Score)
}

set.seed(13579)  # make it completely reproducible
x = rnorm(20)
y = rnorm(20)

test = lm(y~x)
# getting R2 it from the model
summary(test)
y_pred = predict(test)

# calculating it "by hand"
PRESS = sum((y - y_pred)^2)
SS = sum((y - mean(y))^2)
1 - (PRESS/SS)

# using the R2_Score function
R2_Score(y_pred, y)  
```

Just to press the point made in https://github.com/yanyachen/MLmetrics/issues/13#issuecomment-1069165888, make sure to use the correct order of parameters.

The output from the code above is:

```
# getting R2 from the model
Call:
lm(formula = y ~ x)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.83855 -0.96018  0.00419  0.84079  1.64995 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)
(Intercept)  0.15186    0.24524   0.619    0.544
x            0.05112    0.24861   0.206    0.839

Residual standard error: 1.082 on 18 degrees of freedom
Multiple R-squared:  0.002343,	Adjusted R-squared:  -0.05308 
F-statistic: 0.04227 on 1 and 18 DF,  p-value: 0.8394`

# calculating it "by hand"
[1] 0.002343091

# using the R2_Score function
[1] 0.002343091
```

I am using R 4.4.3 under Windows 11 and Ubuntu Linux (tested on both OS)

*Bottom line*: Perhaps the problem was not using the parameters in the correct order when calling the function.

**Comment by fionag11** [comment]
Thanks for running through that. You are right, in regards to this example. However, I have some data that have yielded problematic results when calculating R squared in this way.  I think this ties into a bigger problem related to the calculation of R-squared as discussed in[ [Kvalseth (1985](https://www.tandfonline.com/doi/abs/10.1080/00031305.1985.10479448))
More later when I have a chance to put together  my examples in a meaningful script.
